hawkular-tenant-jaxrs-filter/src/main/java/org/hawkular/jaxrs/filter/tenant/TenantFeature.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2014-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.jaxrs.filter.tenant;
+
+import java.lang.reflect.Method;
+
+import javax.ws.rs.container.DynamicFeature;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * @author Juraci Paixão Kröhling
+ */
+@Provider
+public class TenantFeature implements DynamicFeature {
+    @Override
+    public void configure(ResourceInfo resourceInfo, FeatureContext context) {
+        Class<?> resourceClass = resourceInfo.getResourceClass();
+        Method method = resourceInfo.getResourceMethod();
+
+        boolean required = true;
+        if (resourceClass.isAnnotationPresent(TenantRequired.class)) {
+            required = resourceClass.getAnnotation(TenantRequired.class).value();
+        }
+
+        if (method.isAnnotationPresent(TenantRequired.class)) {
```

Comment:
Yes. That's for the case where all methods on a class don't require the Hawkular-Tenant header, except one: instead of marking all methods with "false", the class is marked with "false" and a single method is marked with "true". This behavior is in line with similar Java EE annotations: the most specific setting is always the one which is effective.